alf/networks/encoding_networks.py

```diff
@@ -803,7 +803,7 @@ def forward(self, inputs, state=()):
         # call super to preprocess inputs
         z, state = super().forward(inputs, state, max_outer_rank=2)
         if self._img_encoding_net is None and len(self._fc_layers) == 0:
-            if inputs.ndim == 2:
+            if len(alf.nest.get_nest_shape(inputs)) == 2:
```

Comment:
This check is not general enough. Imagine that I have several inputs but get_nest_shape will return the shape of an arbitrary input.

Reply:
I wonder what would be a good solution here.  Using any field that's not explicitly specified would be risky?

Reply:
ImageEncodingNetwork takes only one input

Reply:
Using z.ndim in the end as suggested by Wei.  @hnyu let me know if you have a better idea.